interop-testing/src/test/java/io/grpc/testing/integration/DualStackClientTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.testing.integration;
+
+import static io.grpc.testing.integration.Util.AddressType.IPV4;
+import static io.grpc.testing.integration.Util.AddressType.IPV4_IPV6;
+import static io.grpc.testing.integration.Util.AddressType.IPV6;
+import static org.junit.Assert.assertEquals;
+
+import io.grpc.Channel;
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.TlsChannelCredentials;
+import io.grpc.testing.GrpcCleanupRule;
+import io.grpc.testing.TlsTesting;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link StressTestClient}. */
+@RunWith(JUnit4.class)
+public class DualStackClientTest {
+  protected static final EmptyProtos.Empty EMPTY = EmptyProtos.Empty.getDefaultInstance();
+
+  @Rule
+  public final GrpcCleanupRule cleanupRule = new GrpcCleanupRule();
+
+  @Rule
+  public final Timeout globalTimeout = Timeout.seconds(10);
+
+
+  @Test
+  public void check_ipv4() throws Exception {
+    checkConnectionWorks(IPV4);
+  }
+
+  @Test
+  public void check_ipv6() throws Exception {
+    checkConnectionWorks(IPV6);
+  }
+
+  @Test
+  public void check_ipv4_ipv6() throws Exception {
+    checkConnectionWorks(IPV4_IPV6);
+  }
+
+  @Test
+  public void checkNoAddressType() throws Exception {
+    checkConnectionWorks(null);
+  }
+
+  private void checkConnectionWorks(Util.AddressType addressType)
+      throws Exception {
+
+    TestServiceServer server = getAndStartTestServiceServer(addressType);
+    String target = getTargetServer(addressType);
```

Comment:
How does changing the target string between the tests prove anything? It seems we should use `localhost` unconditionally here, expecting localhost maps to `::1` and `127.0.0.1`. We can have an Assume check that ignores these tests if it doesn't map to at least one IPv4 and one IPv6?

Reply:
Discussed offline. We are testing the server, not the client (thus the test class rename). Thus we want different ips.